application/github/v0/pull_request.go

```diff
@@ -0,0 +1,174 @@
+	"github.com/instill-ai/x/errmsg"
+	"google.golang.org/protobuf/types/known/structpb"
+)
+
+type PullRequestService interface {
+	List(context.Context, string, string, *github.PullRequestListOptions) ([]*github.PullRequest, *github.Response, error)
+	Get(context.Context, string, string, int) (*github.PullRequest, *github.Response, error)
+	ListComments(context.Context, string, string, int, *github.PullRequestListCommentsOptions) ([]*github.PullRequestComment, *github.Response, error)
+	CreateComment(context.Context, string, string, int, *github.PullRequestComment) (*github.PullRequestComment, *github.Response, error)
+	ListCommits(context.Context, string, string, int, *github.ListOptions) ([]*github.RepositoryCommit, *github.Response, error)
+}
+
+type PullRequest struct {
+	ID          		int64           	`json:"id"`
+	Number      		int             	`json:"number"`
+	State       		string          	`json:"state"`
+	Title	 			string 				`json:"title"`
+	Body 				string 				`json:"body"`
+	DiffURL 			string 				`json:"diff_url,omitempty"`
+	CommitsURL 			string 				`json:"commits_url,omitempty"`
+	Commits 			[]Commit 			`json:"commits"`
+	Head 				string 				`json:"head"`
+	Base 				string 				`json:"base"`
+	CommentsNum 		int 				`json:"comments_num"`
+	CommitsNum 			int 				`json:"commits_num"`
+	ReviewCommentsNum 	int		 			`json:"review_comments_num"`
+}
+func (githubClient *Client) extractPullRequestInformation(originalPr *github.PullRequest) (PullRequest, error) {
+	resp := PullRequest{
+		ID: originalPr.GetID(),
+		Number: originalPr.GetNumber(),
+		State: originalPr.GetState(),
+		Title: originalPr.GetTitle(),
+		Body: originalPr.GetBody(),
+		DiffURL: originalPr.GetDiffURL(),
+		Head: originalPr.GetHead().GetSHA(),
+		Base: originalPr.GetBase().GetSHA(),
+		CommentsNum: originalPr.GetComments(),
+		CommitsNum: originalPr.GetCommits(),
+		ReviewCommentsNum: originalPr.GetReviewComments(),
+	}
+	if originalPr.GetCommitsURL() != "" {
+		commits, _, err := githubClient.client.PullRequests.ListCommits(context.Background(), githubClient.owner, githubClient.repository, resp.Number, nil)
+		if err != nil {
+			return PullRequest{}, err
+		}
+		resp.Commits = make([]Commit, len(commits))
+		for idx, commit := range commits {
+			resp.Commits[idx] = githubClient.extractCommitInformation(commit)
+		}
+	}
+	return resp, nil
+}
+
+// for testing
+type GetAllPullRequestsInput struct {
+	Owner		 	string 			`json:"owner"`
+	Repository		string 			`json:"repository"`
+	State		 	string 			`json:"state"`
+	Sort		 	string 			`json:"sort"`
+	Direction		string 			`json:"direction"`
+}
+type GetAllPullRequestsResp struct {
+	PullRequests   	[]PullRequest 	`json:"pull_requests"`
+}
+
+func (githubClient *Client) getAllPullRequestsTask(props *structpb.Struct) (*structpb.Struct, error) {
+	err := githubClient.setTargetRepo(props)
+	if err != nil {
+		return nil, err
+	}
+
+	opts := &github.PullRequestListOptions{
+		State: props.GetFields()["state"].GetStringValue(),
```

Comment:
Important
I think making dependency clean is important.

Could you take a look https://github.com/instill-ai/component/pull/187#discussion_r1662946437?

TLDR
I want to make protobuf struct to component struct.
In our other services, we depend on component struct rather than protobuf struct.

Reply:
Got it! I have made all tasks to component struct